client/src/main/java/io/pravega/client/stream/impl/TransactionalEventStreamWriterImpl.java

```diff
@@ -198,13 +199,19 @@ private void throwIfClosed() throws TxnFailedException {
 
     @Override
     public Transaction<Type> getTxn(UUID txId) {
-        StreamSegments segments = getAndHandleExceptions(
-                controller.getCurrentSegments(stream.getScope(), stream.getStreamName()), RuntimeException::new);
+        // check if the transaction is open.
         Status status = getAndHandleExceptions(controller.checkTransactionStatus(stream, txId), RuntimeException::new);
         if (status != Status.OPEN) {
             return new TransactionImpl<>(writerId, txId, controller, stream);
         }
-        
+
```

Comment:
question: why do we need to get the segments for a txn again? when you create a txn, controller returns the txn segments, right?

so when is this getTxn api useful?

Reply:
`TransactionalEventStreamWriter` enables users to created transactions. The user can use 

1. `TransactionalEventStreamWriter#beginTxn()` to create a new transaction
2. `TransactionalEventStreamWriter#getTxn(UUID txnId)` to fetch an already created transaction given its id. This is useful when the user knows a txnId and wants to perform operations on that transaction.

Both the above APIs return a `io.pravega.client.stream.Transaction` using which the user can 
a. Write events to a transaction, `Transaction#writeEvent`
b. Commit a transaction, `Transaction#commit() `
c. Abort a transaction, `Transaction#abort()`
d. Check the status of the transaction `Transaction#checkStatus()`.

The new controller API ensures the correct segments are fetched when the user uses  [2.] getTxn(txnId) API.